tools/bug-report/pkg/archive/archive.go

```diff
@@ -0,0 +1,111 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package archive
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	proxyLogsPathSubdir = "proxy-logs"
+	istioLogsPathSubdir = "istio-logs"
+	clusterInfoSubdir   = "cluster"
+)
+
+var (
+	// Each run of the command produces a new archive.
+	instancePath = fmt.Sprint(rand.Int())
```

Comment:
I think go can make a random dir for you as TempDir or something

Reply:
can't really do that here because the actual dir creation is deferred till just before writing the file. since i'm doing essentially mkdir -p before writing every file due to the dynamic file structure, there's not much point creating the base temp dir at the start.

Reply:
`TempDir` is more reliable.

Reply:
i forgot about cross-platform which tempdir will take care of - will change this.